docs/service_architecture.md

```diff
@@ -12,12 +12,63 @@ weight=3
 
 # Understand the Notary service architecture
 
-On this page, you get an overview of the Notary service architecture. This
-document assumes a prior understanding of [The Update
-Framework](https://theupdateframework.github.io/).
+On this page, you get an overview of the Notary service architecture.
+
+## Brief overview of TUF keys and roles
+
+This document assumes familiarity with [The Update Framework](https://theupdateframework.github.io/),
+but here is a brief recap of the TUF roles and corresponding key hierarchy:
+
+<center><img src="images/key-hierarchy.svg" alt="TUF Key Hierarchy" width="400px"/></center>
+
+- The root key is the root of all trust.  It signs the
+[root metadata file](https://github.com/theupdateframework/tuf/blob/develop/docs/tuf-spec.txt#L489),
+which lists the IDs of the root, targets, snapshot, and timestamp public keys.
+Clients use these public keys to verify the signatures on all the metadata files
+in the repository.  This key is held by a collection owner, and should be kept offline
+and safe, more so than any other key.
+
+- The snapshot key signs the
+[snapshot metadata file](https://github.com/theupdateframework/tuf/blob/develop/docs/tuf-spec.txt#L604),
+which enumerates the filenames, sizes, and hashes of the root,
+targets, and delegation metadata files for the collection.  This file is used to
```

Comment:
nit: We standardize on a *single* space after full stops, so one space before "This file"